### PR TITLE
chore: audit refactor component

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,3 +1,3 @@
-apiBaseUrl=
-sdkBaseUrl=
-mixpanelToken=
+apiBaseUrl=https://sandbox.hyperswitch.io
+sdkBaseUrl=https://beta.hyperswitch.io/v1/HyperLoader.js
+mixpanelToken=
```
Comment (PritishBudhiraja) [inline .env:3]: .env removal

File: config/FeatureFlag.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "production_access": false,
   "quick_start": true,
   "switch_merchant": true,
-  "audit_trail": false,
+  "audit_trail": true,
```
Comment (PritishBudhiraja) [inline config/FeatureFlag.json:8]: Do we need it to be true in all environments? And by default too?

File: src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res [hunk 3]
```diff
@@ -1,151 +1,19 @@
...
-      </Section>
-    }}
-  </PageLoaderWrapper>
+  <AuditLogUI
+    id={paymentId}
+    promiseArr={[
+      fetchDetails(disputesLogsUrl),
+      fetchDetails(webhooksLogsUrl),
+      fetchDetails(connectorLogsUrl),
+    ]}
```
Comment (PritishBudhiraja) [inline src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res:16]: Can't we move all this API Call to AuditLogUI as nothing happening in this component except passing the prop

File: src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res [hunk 4]
```diff
@@ -1,151 +1,19 @@
...
-  let (selectedOption, setSelectedOption) = React.useState(_ => {
-    value: 0,
-    optionType: API_EVENTS,
-  })
-  let isError = React.useMemo0(() => {ref(false)})
-  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
 
   let disputesLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/api_event_logs?type=Dispute&payment_id=${paymentId}&dispute_id=${disputeId}`
   let webhooksLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/outgoing_webhook_event_logs?&payment_id=${paymentId}&dispute_id=${disputeId}`
   let connectorLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/connector_event_logs?payment_id=${paymentId}&dispute_id=${disputeId}`
```
Comment (PritishBudhiraja) [inline src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res:9]: Please add these URLs in APIUtils

File: src/screens/Analytics/Logs/RefundLogs/RefundLogs.res [hunk 5]
```diff
@@ -1,151 +1,19 @@
...
-  let (selectedOption, setSelectedOption) = React.useState(_ => {
-    value: 0,
-    optionType: API_EVENTS,
-  })
-  let isError = React.useMemo0(() => {ref(false)})
-  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
 
   let refundsLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/api_event_logs?type=Refund&payment_id=${paymentId}&refund_id=${refundId}`
   let webhooksLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/outgoing_webhook_event_logs?&payment_id=${paymentId}&refund_id=${refundId}`
   let connectorLogsUrl = `${HSwitchGlobalVars.hyperSwitchApiPrefix}/analytics/v1/connector_event_logs?payment_id=${paymentId}&refund_id=${refundId}`
```
Comment (PritishBudhiraja) [inline src/screens/Analytics/Logs/RefundLogs/RefundLogs.res:8]: Same APIUtils.res

File: src/screens/Analytics/Logs/LogUtils/LogUtils.res [hunk 6]
```diff
@@ -5,3 +5,117 @@ let sortByCreatedAt = (log1: JSON.t, log2: JSON.t) => {
...
+  | PaymentsUpdate
+  | RefundsCreate
+  | RefundsUpdate
+  | DisputesEvidenceSubmit
+  | AttachDisputeEvidence
+  | RetrieveDisputeEvidence
+  | IncomingWebhookReceive
+  | NotDefined
+
+let itemToObjMapper = flowString => {
```
Comment (PritishBudhiraja) [inline src/screens/Analytics/Logs/LogUtils/LogUtils.res:24]: name can be more specific of what it's doing

**Review by JeevaRamu0104 — COMMENTED** [review]
try adding test cases

File: src/screens/Analytics/Logs/LogUtils/LogUtils.res (hunk 6 above)
Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/LogUtils/LogUtils.res:24]: this is the std name we use for string to object converter

File: src/screens/Analytics/Logs/RefundLogs/RefundLogs.res (hunk 5 above)
Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/RefundLogs/RefundLogs.res:8]: Current util file doesn't support passing more than one id in url

File: src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res (hunk 4 above)
Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res:9]: Current util file doesn't support passing more than one id in url

File: src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res (hunk 3 above)
Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/DisputeLogs/DisputeLogs.res:16]: functions can also can be post or get so its been passed from parent componet itself

File: .env (hunk 1 above)
Comment (sagarnaikjuspay) [reply .env:3]: updated

File: config/FeatureFlag.json (hunk 2 above)
Comment (sagarnaikjuspay) [reply config/FeatureFlag.json]: no, updated

File: src/screens/Analytics/Logs/LogUtils/LogUtils.res (hunk 6 above)
Comment (PritishBudhiraja) [reply src/screens/Analytics/Logs/LogUtils/LogUtils.res:24]: can be named as **flowTypeStringToTypeMapper**

File: src/screens/Analytics/Logs/LogUtils/AuditLogUI.res [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+    let isValidNonEmptyValue = value => {
+      switch value->JSON.Classify.classify {
+      | Bool(_) => true
+      | String(_) => true
+      | Number(_) => true
+      | Object(_) => true
+      | _ => false
+      }
+    }
```
Comment (Riddhiagrawal001) [inline src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:21]:
```suggestion
    let isValidNonEmptyValue = value => {
      switch value->JSON.Classify.classify {
      | Bool(_) | String(_) | Number(_) | Object(_) => true
      | _ => false
      }
    }
```

File: src/screens/Analytics/Logs/LogUtils/AuditLogUI.res [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+          {`No logs available for this ${(logType :> string)->String.toLowerCase}`->React.string}
+        </div>
+      </div>
+    } else {
+      <Section
+        customCssClass={`bg-white dark:bg-jp-gray-lightgray_background rounded-md pt-2 pb-4 flex gap-7 justify-between h-48-rem !max-h-50-rem !min-w-[55rem] max-w-[72rem] overflow-scroll`}>
+        {timeLine}
+        {codeBlock}
+      </Section>
+    }}
```
Comment (Riddhiagrawal001) [inline src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:209]: Can we put this if else under UIUtils.RenderIf

File: src/screens/Analytics/Logs/LogUtils/LogUtils.res [hunk 9]
```diff
@@ -5,3 +5,134 @@ let sortByCreatedAt = (log1: JSON.t, log2: JSON.t) => {
...
+      eventDict->getString("component", "")->JSON.Encode.string,
+    )
+    eventDict->Dict.set("customer_device", eventDict->getString("platform", "")->JSON.Encode.string)
+    eventDict->Dict.set("sdk_version", eventDict->getString("version", "")->JSON.Encode.string)
+    eventDict->Dict.set(
+      "event_name",
+      updatedEventName
+      ->snakeToTitle
+      ->titleToSnake
+      ->snakeToCamel
```
Comment (Riddhiagrawal001) [inline src/screens/Analytics/Logs/LogUtils/LogUtils.res:123]: Can be directly changed to `snakeToTitle->snakeToCamel`

File: src/screens/Analytics/Logs/LogUtils/AuditLogUI.res (hunk 8 above)
Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:210]: UIUtils.RenderIf returns React.null on false condition but here we want to show something in that case

Comment (Riddhiagrawal001) [reply src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:207]: we can put two conditions one for the `true` case and the other one for `!true` case

let checkCondition = id->HSwitchOrderUtils.isTestData || data->Array.length === 0

<UIUtils.RenderIf condition = {checkCondition}>
{some react component}
</UIUtils.RenderIf>


<UIUtils.RenderIf condition = {!checkCondition}>
{some react component}
</UIUtils.RenderIf>

Comment (sagarnaikjuspay) [reply src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:207]: not necessary ig, it won't to good practice

Comment (Riddhiagrawal001) [reply src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:207]: Thats generally the practice we follow in dashboard .

File: src/screens/Analytics/Logs/PaymentLogs/PaymentLogsUtils.res
Comment (JeevaRamu0104) [inline src/screens/Analytics/Logs/PaymentLogs/PaymentLogsUtils.res:1]: Is this file being used?

File: src/screens/Analytics/Logs/LogUtils/AuditLogUI.res [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+          switch arr->Array.get(0) {
+          | Some(dict) =>
+            switch dict->getDictFromJsonObject->getLogType {
+            | SDK => logs->Array.pushMany(arr->parseSdkResponse)->ignore
+            | CONNECTOR | API_EVENTS => logs->Array.pushMany(arr)->ignore
+            | WEBHOOKS => logs->Array.pushMany([dict])->ignore
+            }
+          | _ => ()
+          }
+        | String(_) => isError.contents = true
```
Comment (JeevaRamu0104) [inline src/screens/Analytics/Logs/LogUtils/AuditLogUI.res:103]: please recheck this.Even if single value is string your setting isError has true.

**Review by JeevaRamu0104 — DISMISSED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by gitanjli525 — APPROVED** [review]

**Pull request merged**